File: src/main/java/snippets/classes/java/Rectangle.java
package snippets.classes.java;

public class Rectangle {

  private int x;
  private int y;
  private int width;
  private int height;

  Rectangle(int x, int y, int width, int height) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
  }

  // getters
  // setters
  // hashcode / equals

  public int getArea() {
    return width * height;
  }

  public int getPerimeter() {
    return width * 2 + height * 2;
  }

  public static void main(String[] args) {
    Rectangle rectangle = new Rectangle(0, 0, 10, 20);
  }
}

File: src/main/java/snippets/classes/kotlin/Rectangle.kt
package snippets.classes.kotlin

class Rectangle(var x: Int, var y: Int, var width: Int, var height: Int) {
  val area get() = width * height
  val perimeter get() = width * 2 + height * 2
}

val rectangle = Rectangle(0, 0, 10, 20)
